Add Coala tests in Circle CI.

matejak opened the pull request:
This PR aims at getting the new Coala tests to work, while keeping the tox tests working.
Aimed to fix #34

matejak committed:
Add Coala tests in Circle CI.

matejak committed:
Perform some simple style fixes

matejak committed:
Fix tiny formatting issues

matejak commented:
The checkers are upset about

- unused variables (I agree with that, leaving those feels unfinished, and a pull request should be a finished piece of work), and
- lines that are too long. As it is stated in the [talk](https://www.youtube.com/watch?v=wf-BqAjZb8M), it is difficult to adhere to the 80-char limit without violating other recommendations or best practices.

We can postpone this PR until we meet again on the mentoring session, or a dictator can impose some kind of solution.

petrjoachim commented:
I think it would be ok to have 120 char limit, but I still don't like it 🤔 
For the unused (but prepared  variables, it should be ok to put them in a comment or make an "extraction" function to get them from request to a documented structure (for instance)

xmalmed committed:
Add estetical changes.

The post handling will change soon, so no big deal.

xmalmed commented:
I don't understand, what is bad about *tests/unit/test_start.py* :(

iakat commented:
coala is always written with a lower case c. 😧

iakat commented:
https://wrongsideofmemphis.wordpress.com/2013/03/25/80-chars-per-line-is-great/

There is no such thing as "the talk" regarding PEP8 controversial rules.

@petrjoachim can we go for 80 chars max. here? This is not a very complex project, either.

matejak committed:
Remove the PyRoma bear

The need of having a project's e-mail address set up
is a n unneccessary burden at this stage of the project.

matejak committed:
Remove the unused pytest import

matejak commented on src/iwant_bot/start.py:
```diff
@@ -79,10 +79,7 @@ def check_post_request(body):
         return res
-        #return web.json_response(body=format_response(user_name + ' wants me to ' + text))
     elif trigger_word == 'whoami':
         return web.json_response(body=format_response('You are ' + user_name + ' with id ' + user_id))
```
Actually this line doesn't have to be so long - I would propose:
- Usage of formatted strings s.a. `f'You are {user_name} with id {user_id}'`.
- Having two statements (on two lines) instead of a single one - `response = f'You are ...'`, `return web.json_response(body=format_response(response))`

xmalmed committed:
Prune the code to be more aesthetic.

Set max_line_length to 80.
Remove commented lines.
Reduce lines length.

iakat commented on .coafile:
```diff
@@ -0,0 +1,4 @@
+[python]
+bears = PEP8Bear,PyFlakesBear,PySafetyBear,RadonBear
+max_line_length = 99
```
80, agreed with Jiri

iakat commented on circle.yml:
```diff
@@ -1,3 +1,10 @@
+
+test:
+  override:
+    - docker run -v=$(pwd):/app --workdir=/app coala/base coala --ci    
```
tag coala verison `coala/base:0.11`

iakat commented on setup.py:
```diff
@@ -96,4 +96,5 @@
             'sample=sample:main',
         ],
     },
+    zip_safe=True,
```
out of context for this commit

iakat replied:
^^^^

iakat replied:
explain why this is here @matejak

matejak replied:
I have explained it on Slack, but whatever: Concerning the `zip_safe` thing, I recall that we have agreed to keep it in the PR on Friday. By the way, it actually was relevant for the PR (as the `PyRoma` bear was complaining about it).

iakat commented on .coafile:
```diff
@@ -1,4 +1,3 @@
 [python]
 bears = PEP8Bear,PyFlakesBear,PySafetyBear,PyromaBear,RadonBear
 files = **.py
-
```
empty line at end of file is good

iakat replied:
https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-file

iakat commented on src/iwant_bot/start.py:
```diff
@@ -59,16 +59,16 @@ def check_post_request(body):
+        # service_id =    body["service_id"]
+        # channel_id =    body["channel_id"]
+        # channel_name =  body["channel_name"]
+        # timestamp =     body["timestamp"]
```
why is this commented? can it be removed?

matejak committed:
Add a newline at the end of the coafile

iakat commented on src/iwant_bot/start.py:
```diff
@@ -11,8 +11,8 @@ def add_numbers(a, b):
-   ret = f"'{message.text}' by '{message.nickname}' at {message.when}"
-   return ret
+    ret = f"'{message.text}' by '{message.nickname}' at {message.when}"
+    return ret
```
return f"'{message.text}' by '{message.nickname}' at {message.when}"

iakat commented on src/iwant_bot/start.py:
```diff
@@ -54,18 +57,10 @@ def check_post_request(body):
-        user_id =       body["user_id"]
-        user_name =     body["user_name"]
-        text =          body["text"]
+        user_id = body["user_id"]
```
" -> '

iakat commented on src/iwant_bot/start.py:
```diff
@@ -75,16 +70,14 @@ def check_post_request(body):
-        print(res.body)  # how to display full response body? It is <aiohttp.payload.StringPayload object at 0x7fc9e266c2e8>
-        return res
-        #return web.json_response(body=format_response(user_name + ' wants me to ' + text))
+        message = format_response(user_name + ' wants me to ' + text)
```
let's use f-strings

iakat commented on src/iwant_bot/start.py:
```diff
@@ -75,16 +70,14 @@ def check_post_request(body):
     elif trigger_word == 'whoami':
-        return web.json_response(body=format_response('You are ' + user_name + ' with id ' + user_id))
+        message = format_response(
+            'You are ' + user_name + ' with id ' + user_id)
```
f-strings

iakat commented on src/iwant_bot/start.py:
```diff
@@ -25,27 +25,30 @@ def format_message(message):
 
-    expected_fields = {"token", "team_id", "team_domain", "service_id", "channel_id",
-                       "channel_name", "timestamp", "user_id", "user_name", "text"}
+    expected_fields = {"token", "team_id", "team_domain", "service_id",
```
why is this a dict and not a tuple?

xmalmed replied:
It's a set.

xmalmed committed:
Replace strings by f-strings and unify single and double quotes.

xmalmed commented:
Can you define the convention for ' and " usage?
I just found, that someone try to use " for the strings read by people (system messages, outputs, etc.) and ' for strings inside the code only (like keys of dict). Do you agree?

iakat commented:
use ' unless you need to escape it, in that case use ", that's our convention and it's nice

iakat commented on src/iwant_bot/db.py:
```diff
@@ -47,15 +47,16 @@ def add_new_nickname_to_session(self, session, nickname):
         try:
-            query = session.query(sqlalchemy.exists().where(Nickname.name == nickname))
+            query = session.query(
+                sqlalchemy.exists().where(Nickname.name == nickname))
```
move ending ) to newline

iakat commented on src/iwant_bot/start.py:
```diff
@@ -25,27 +24,29 @@ def format_message(message):
 def format_response(message="No response."):
-    ret = f'{{\n\t"text":"{message}"\n}}\n'
-    return ret
+    return f'{{\n\t"text":"{message}"\n}}\n'
```
use the json or simplejson library here please

iakat commented on src/iwant_bot/start.py:
```diff
@@ -25,27 +24,29 @@ def format_message(message):
 
-    expected_fields = {"token", "team_id", "team_domain", "service_id", "channel_id",
-                       "channel_name", "timestamp", "user_id", "user_name", "text"}
+    expected_fields = {'token', 'team_id', 'team_domain', 'service_id',
```
why is this a dict and not a tuple, part 2

xmalmed replied:
@chauffer I have already replied in the part 1.
>  It's a set.

iakat replied:
yeah, my bad, why is it not a tuple

xmalmed replied:
@chauffer: Because set operations (difference) is elegant and readable.

iakat commented on src/iwant_bot/start.py:
```diff
@@ -54,37 +55,26 @@ def check_post_request(body):
-        print(res.body)  # how to display full response body? It is <aiohttp.payload.StringPayload object at 0x7fc9e266c2e8>
-        return res
-        #return web.json_response(body=format_response(user_name + ' wants me to ' + text))
+        message = format_response(f"{user_name} wants me to {text}")
```
" -> '

iakat commented on src/iwant_bot/start.py:
```diff
@@ -54,37 +55,26 @@ def check_post_request(body):
+        message = format_response(f"{user_name} wants me to {text}")
     elif trigger_word == 'whoami':
-        return web.json_response(body=format_response('You are ' + user_name + ' with id ' + user_id))
+        message = format_response(f"You are {user_name} with id {user_id}")
```
" -> '

iakat commented on src/iwant_bot/start.py:
```diff
@@ -54,37 +55,26 @@ def check_post_request(body):
+        message = format_response(f"You are {user_name} with id {user_id}")
     else:
-        return web.json_response(body=format_response(user_name + ' wrote ' + text))
+        message = format_response(f"{user_name} wrote {text}")
```
" -> '

iakat commented on src/iwant_bot/start.py:
```diff
@@ -25,27 +24,29 @@ def format_message(message):
-    text += ['> curl -X POST  --data "user_name=0&channel_id=1&channel_name=2&service_id=3&']
-    text += ['  team_domain=4&team_id=5&text=6&timestamp=7&token=9&user_id=9" http://localhost:8080']
-             #curl -X POST some_url_like_https://kiwislackbot1.localtunnel.me/\n']
+    text += ['> curl -X POST  --data "user_name=0&channel_name=2']
```
this changes the data in it, out of context for this MR

iakat replied:
this changes the data in it, out of context for this MR

xmalmed replied:
@chauffer:  I need to change the example anyway, because the lines are to long. So, I decided to simplified example and keep it on 2 lines.

iakat replied:
ok, so you can decide

1. squash the commits yourself to multiple changes, so we don't have to do it on github

2. out of commit for this MR

xmalmed committed:
Unify the quotes and move ending ) to newline.

iakat commented on src/iwant_bot/db.py:
```diff
@@ -52,7 +52,8 @@ def save_message(self, message, nickname):
             query = session.query(
-                sqlalchemy.exists().where(Nickname.name == nickname))
+                sqlalchemy.exists().where(Nickname.name == nickname)
+                )
```
deindent

xmalmed replied:
@chauffer I found all three possibilities in PEP8, even the ) at the end of first line. Please, if you mention an internal agreement, can you be more specific?

iakat commented on src/iwant_bot/start.py:
```diff
@@ -45,8 +45,8 @@ def check_post_request(body):
-        print("POST missing these fields: "
-              ", ".join(expected_fields - obtined_fields))
+        print('POST missing these fields: '
+              ', '.join(expected_fields - obtined_fields))
```
same ) story

iakat commented on .coafile:
```diff
@@ -0,0 +1,5 @@
+[python]
+bears = PEP8Bear,PyFlakesBear,PySafetyBear,RadonBear
```
`,` -> `, `

iakat replied:
add space afterr comma

xmalmed committed:
Add or delete aesthetic spaces.

petrjoachim reviewed (approved):
zip_safe was explained.
